data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/S3InputStream.java

```diff
@@ -668,6 +668,8 @@ private void recordS3Exception(final S3Exception ex) {
             s3ObjectPluginMetrics.getS3ObjectsFailedNotFoundCounter().increment();
         } else if (ex.statusCode() == HttpStatusCode.FORBIDDEN) {
             s3ObjectPluginMetrics.getS3ObjectsFailedAccessDeniedCounter().increment();
+        } else if (ex.statusCode() == HttpStatusCode.THROTTLING) {
+            s3ObjectPluginMetrics.getS3ObjectsThrottledCounter().increment();
```

Comment:
Do we really need the throttling metric? This can come from the AWS SDK if we wire the metrics up for Micrometer.